### PR TITLE
[MNG-7010] Omit "NB: JAVA_HOME should point to a JDK not a JRE"

### PR DESCRIPTION
**Pull request by michael-o**
This closes #512

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[MNG-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MNG-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

File: apache-maven/src/assembly/shared/init [hunk 1]
```diff
@@ -50,14 +50,15 @@ if [ -z "$JAVACMD" ] ; then
...
-  echo "This environment variable is needed to run this program" >&2
-  echo "NB: JAVA_HOME should point to a JDK not a JRE" >&2
+  echo "The JAVA_HOME environment variable is not defined correctly," >&2
+  echo "this environment variable is needed to run this program." >&2
   exit 1
 fi
 
 if [ -z "$JAVA_HOME" ] ; then
   echo "Warning: JAVA_HOME environment variable is not set." >&2
+elif [ ! -x "$JAVA_HOME/bin/javac" ] ; then
```
Comment (elharo) [inline apache-maven/src/assembly/shared/init:60]: Now that I look at this I see a problem. This is testing that JAVA_HOME does not point to a JDK. It is not testing that it does point to a JRE. The problem I encountered when I filed the bug was that JAVA_HOME pointed to a directory that did not exist. Maybe test also that $JAVA_HOME/bin/java does exist?

Comment (michael-o) [reply apache-maven/src/assembly/shared/init:60]: Isn't it done with `if [ ! -x "$JAVACMD" ] ; then`?

**Comment by michael-o** [comment]
Something is wrong the Java 8 on all runners. `JAVA_HOME` isn't set. Can we fix this ourselves or GitHub support?

File: apache-maven/src/assembly/shared/init (hunk 1 above)
Comment (elharo) [reply apache-maven/src/assembly/shared/init:60]: Honestly not sure. Bash is not my strong suit.

Comment (michael-o) [reply apache-maven/src/assembly/shared/init:60]: Technically, Robert's right on JIRA. Maven does not require a JDK, plugins do.

**Comment by michael-o** [comment]
@elharo  I have simplified the PR. As mentioned by @rfscholte we should improve the remaining text in a different issue/PR.

**Comment by jira-importer** [comment]
Resolve #8026

**Pull request closed without merging**